addons/project/models/project_update.py

```diff
@@ -172,14 +130,18 @@ def _get_last_updated_milestone(self, project):
                       WHERE mm.model = 'project.milestone'
                         AND mm.message_type = 'notification'
                         AND pm.project_id = %(project_id)s
-                        AND mm.date > (now() at time zone 'utc')::date - '1 month'::interval
+         """
+        if project.last_update_id.date:
+            query = query + "AND mm.date <= %(last_update_date)s"
+        query = query + """
                      WINDOW w_partition AS (
                              PARTITION BY pm.id
                              ORDER BY mm.date ASC
                             )
-                   ORDER BY pm.deadline ASC;
+                   ORDER BY pm.deadline ASC
+                   LIMIT 1;
```

Comment:
This is what we originally had, but I had to change it because I got a warning on the runbot telling me that the risk of sql injection was high with such a request, so I changed it a bit.